src/features/embalm/stepContent/hooks/useCreateSarcohpagus.ts

```diff
@@ -31,103 +52,183 @@ export function useCreateSarcophagus() {
   const { uploadArweaveFile } = useArweaveService();
-  const { submitSarcophagus } = useSubmitSarcophagus();
-
-  // TODO: Move this into its own hook and check all fields
-  // TODO: render a message to the user about what data is missing
-  const canCreateSarcophagus = recipientState.publicKey !== '' && !!outerPublicKey && !!file;
-
-  const uploadAndSetDoubleEncryptedFile = useCallback(async () => {
-    const emptyData = { sarcophagusPayloadTxId: '' };
-
-    if (!canCreateSarcophagus) return emptyData;
-
-    const payload = await readFileDataAsBase64(file);
+  const { dialSelectedArchaeologists, initiateSarcophagusNegotiation } = useSarcophagusNegotiation();
+  const [currentStep, setCurrentStep] = useState(CreateSarcophagusStep.NOT_STARTED);
+  const [stepExecuting, setStepExecuting] = useState(false);
+
+  // Create Sarcophagus State
+  const [archaeologistShards, setArchaeologistShards] = useState([] as ArchaeologistEncryptedShard[]);
+  const [encryptedShardsTxId, setEncryptedShardsTxId] = useState('');
+  const [sarcophagusPayloadTxId, setSarcophagusPayloadTxId] = useState('');
+  const [archaeologistSignatures, setArchaeologistSignatures] = useState(new Map<string, string>([]));
+  const [negotiationTimestamp, setNegotiationTimestamp] = useState(0);
+
+  const arweaveTxIds = [sarcophagusPayloadTxId, encryptedShardsTxId];
+
+  const { submitSarcophagus } = useSubmitSarcophagus(
+    {
+      negotiationTimestamp,
+      archaeologistSignatures,
+      archaeologistShards,
+      arweaveTxIds,
+      currentStep
+    }
+  );
 
-    // Step 1: Encrypt the inner layer
-    const encryptedInnerLayer = await encrypt(recipientState.publicKey, payload); // TODO: Restore this line and remove line below when in-app create sarco flow is ready
+  const incrementStep = (): void => {
+    const currentIndex = createSarcophagusSteps.indexOf(currentStep);
+    setCurrentStep(createSarcophagusSteps[currentIndex + 1]);
+  };
 
-    // Step 2: Encrypt the outer layer
-    const encryptedOuterLayer = await encrypt(outerPublicKey, encryptedInnerLayer);
+  const executeStep = async (stepToExecute: Function, ...args: any[]): Promise<any> => {
+    return new Promise((resolve, reject) => {
+      setStepExecuting(true);
 
-    // Step 3: Upload the double encrypted payload to the arweave bundlr
-    const sarcophagusPayloadTxId = await uploadArweaveFile(encryptedOuterLayer); // TODO: change to use uploadFile for Bundlr, once local testing figured out
+      stepToExecute(...args).then((result: any) => {
+        setStepExecuting(false);
 
-    dispatch(setSarcophagusPayloadTxId(sarcophagusPayloadTxId));
-  }, [
-    canCreateSarcophagus,
-    dispatch,
-    file,
-    outerPublicKey,
-    recipientState.publicKey,
-    uploadArweaveFile,
-  ]);
+        // Set current step to next step
+        incrementStep();
+        resolve(result);
+      }).catch((error: any) => {
+        console.log('caught error', error);
+        setStepExecuting(false);
```

Comment:
not using a `finally` here for `setStepExecuting`, as explicitly want `setStepExecuting(false);` to run *before* `incrementStep` in the success case.